fetcher.go

```diff
@@ -77,7 +77,18 @@ func (f *fetcherSession) BlockOfType(ctx context.Context, link ipld.Link, ptype
 	return nb.Build(), nil
 }
 
-func (f *fetcherSession) NodeMatching(ctx context.Context, node ipld.Node, match selector.Selector, cb FetchCallback) error {
+func (f *fetcherSession) nodeMatching(ctx context.Context, initialProgress traversal.Progress, node ipld.Node, match selector.Selector, cb FetchCallback) error {
```

Comment:
It would make more sense to me if this parameter was just named `progress` and the method was named `initialProgress`. `blankProgress` explains the what, but left me wondering why till I dug into the code a bit.